core/plugins/pacemaker.py

```diff
@@ -0,0 +1,73 @@
+import re
+
+from core import checks
+from core.plugintools import PluginPartBase
+from core.cli_helpers import CLIHelper
+
+PACEMAKER_PKGS_CORE = ['pacemaker', 'crmsh', 'corosync']
+PACEMAKER_SVC_EXPR = ['pacemaker[a-zA-Z-]*',
+                      'corosync']
+
+
+class PacemakerStatus(object):
+
+    @property
+    def crm_status(self):
+        return CLIHelper().pacemaker_crm_status
+
+    @property
+    def offline_nodes(self):
+        offline_nodes = []
+        crm_status = self.crm_status()
+        for line in crm_status:
+            regex_match = re.search(r'.*OFFLINE.*\[\s(.*)\s\]',
+                                    line)
+            if regex_match:
+                offline_nodes = regex_match.group(1).split()
+                return offline_nodes
+        return None
+
+    @property
+    def online_nodes(self):
+        online_nodes = []
+        crm_status = self.crm_status()
+        for line in crm_status:
+            regex_match = re.search(r'.*Online.*\[\s(.*)\s\]',
+                                    line)
+            if regex_match:
+                online_nodes = regex_match.group(1).split()
+                return online_nodes
+        return None
+
+
+class PacemakerBase(object):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.status = PacemakerStatus()
+
+
+class PacemakerChecksBase(PacemakerBase, PluginPartBase):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.apt_check = checks.APTPackageChecksBase(
+            core_pkgs=PACEMAKER_PKGS_CORE)
+        svc_exprs = PACEMAKER_SVC_EXPR
+        self.svc_check = checks.ServiceChecksBase(service_exprs=svc_exprs)
+
+    @property
+    def apt_packages_all(self):
+        return self.apt_check.all_formatted
+
+    @property
+    def plugin_runnable(self):
+        return len(self.apt_check.core) > 0
+
+    @property
```

Comment:
There is no need to re-implement these since PacemakerBase() can be used directly from yaml defs.

Reply:
fixed on cab18285d94b5ececa730ae507fe39fc2d2da899